src/features/auth/authSlice.ts

```diff
@@ -0,0 +1,138 @@
+import axiosAuthInstance from "@/libs/axios/axiosAuthInstance"
+import { LoginFormValues } from "@/libs/yup/schemas/loginSchema"
+import { RegisterFormValues } from "@/libs/yup/schemas/registerSchema"
+import {
+  getAccessTokenFromLocalStorage,
+  getRefreshTokenFromLocalStorage,
+  removeTokensFromLocalStorage,
+  saveAccessTokenInLocalStorage,
+  saveRefreshTokenInLocalStorage,
+} from "@/utils/localStorage"
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit"
+import { isAxiosError } from "axios"
+import i18next from "i18next"
+import { toast } from "react-toastify"
+
+export interface AuthState {
+  isAuthenticated: boolean
+  isRegistrationCompleted: boolean
+}
+
+export interface ApiErrorResponse {
+  errors: string[]
+  message: string
+  statusCode: number
+}
+
+const initialState: AuthState = {
+  isAuthenticated: Boolean(
+    getAccessTokenFromLocalStorage() && getRefreshTokenFromLocalStorage(),
+  ),
+  isRegistrationCompleted: false,
+}
+
+export const loginUser = createAsyncThunk(
+  "auth/loginUser",
+  async ({ email, password }: LoginFormValues, thunkAPI) => {
+    try {
+      const { data } = await axiosAuthInstance.post("/login", {
```

Comment:
It is a good practise to keep all your string variables that might be used in multiple places inside one folder. Imagine a situation that enpoint url changes and you would need to search for that string everywhere in the app. By doing a dedicated file for all the api consts you just need to do it in one place